utils/hash_algo_djb2.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2018-2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "hash_algo_djb2.h"
+
+unsigned long hash_algo_djb2(char* str) {
```

Comment:
Maybe we can move the function body of `hash_algo_djb2` to header file `utils/hash_algo_djb2.h` as a `static inline` function since most hash functions are short and suitable for compiler optimizations.